src/ru/nsu/ccfit/zuev/osu/menu/FilterMenu.java

```diff
@@ -22,8 +22,10 @@
 import ru.nsu.ccfit.zuev.osu.helper.InputManager;
 import ru.nsu.ccfit.zuev.osu.helper.StringTable;
 import ru.nsu.ccfit.zuev.osu.helper.TextButton;
+import ru.nsu.ccfit.zuev.osu.helper.DifficultyReCalculator;
 import ru.nsu.ccfit.zuev.osu.menu.SongMenu.SortOrder;
 import ru.nsu.ccfit.zuev.osuplus.R;
+import sun.util.resources.cldr.CalendarData;
 
 public class FilterMenu implements IUpdateHandler, IFilterMenu {
```

Comment:
Please update `loadConfig` method, `saveConfig` method, and the switch statement in line 180 into these so that new filters are stored properly in config (I have tested them).

```java
public void loadConfig(final Context context) {
    final SharedPreferences prefs = PreferenceManager
            .getDefaultSharedPreferences(context);

    final int sortOrder = prefs.getInt("sortorder", 0);
    switch (sortOrder) {
        case 1:
            order = SortOrder.Artist;
            break;
        case 2:
            order = SortOrder.Creator;
            break;
        case 3:
            order = SortOrder.Date;
            break;
        case 4:
            order = SortOrder.Bpm;
            break;
        case 5:
            order = SortOrder.Stars;
            break;
        case 6:
            order = SortOrder.Length;
            break;
        default:
            order = SortOrder.Title;
            break;
    }
    configContext = context;
    setSortText();
}
```

```java
public void saveConfig() {
    if (configContext == null) {
        return;
    }

    final SharedPreferences prefs = PreferenceManager
            .getDefaultSharedPreferences(configContext);
    final SharedPreferences.Editor editor = prefs.edit();

    switch (order) {
        case Artist:
            editor.putInt("sortorder", 1);
            break;
        case Creator:
            editor.putInt("sortorder", 2);
            break;
        case Date:
            editor.putInt("sortorder", 3);
            break;
        case Bpm:
            editor.putInt("sortorder", 4);
            break;
        case Stars:
            editor.putInt("sortorder", 5);
            break;
        case Length:
            editor.putInt("sortorder", 6);
            break;
        default:
            editor.putInt("sortorder", 0);
            break;
    }

    editor.commit();
}
```

Switch statement in line 180:
```java
switch (order) {
    case Title:
        newOrder = SortOrder.Artist;
        break;
    case Artist:
        newOrder = SortOrder.Creator;
        break;
    case Creator:
        newOrder = SortOrder.Date;
        break;
    case Date:
        newOrder = SortOrder.Bpm;
        break;
    case Bpm:
        newOrder = SortOrder.Stars;
        break;
    case Stars:
        newOrder = SortOrder.Length;
        break;
    default:
        newOrder = SortOrder.Title;
        break;
}
```

Reply:
fixed